### DISCUSSION TITLE
Questions around pool size and max overflow

### DISCUSSION THREAD

**Discussion by coredev123** [sqlalchemy/sqlalchemy#10697] [Usage Questions] [answered]
I am very new to this and using an existing application so apologies if this is very basic or missing any detail

I have an application pointing at Superset to take care of the Reporting element and am getting the following common error:
_'QueuePool limit of size <x> overflow <y> reached, connection timed out, timeout <z>'_

As recommended [here](https://docs.sqlalchemy.org/en/13/errors.html#connections-and-transactions), I proceeded to try set the pool size and max overflow.  I am now attempting to pass this down via my applications superset config but am now getting the below error:

_Invalid argument(s) 'pool_size','max_overflow' sent to create_engine(), using configuration PGDialect_psycopg2/NullPool/Engine.  Please check that the keyword arguments are appropriate for this combination of components.
This may be triggered by: 
Issue 1011 - Superset encountered an unexpected error._

I have tried to pass the pool size and max overflow properties across in their own dictionary (commented in screenshot below) but also as part of "connect_args" but regardless are getting this same error.  Is there anything I need to add to allow these arguments to be included in my params to createengine?

![image](https://github.com/sqlalchemy/sqlalchemy/assets/55496683/77d609b5-cf2d-4de8-8693-98a12544fcaa)

**Comment by zzzeek** [comment, marked as answer]
hi -

the error message here: "using configuration PGDialect_psycopg2/NullPool/Engine." indicates that you have also changed your [poolclass](https://docs.sqlalchemy.org/en/20/core/pooling.html#switching-pool-implementations) setting to use `NullPool`.  If you use `NullPool`, which is fine, you've turned off connection pooling, and you will not get any "overflow" errors.  note your original error message "QueuePool limit of size overflow reached, connection timed out, timeout" refers to *QueuePool*, not *NullPool*.

your options are:

a. use NullPool, and dont set max_overflow or pool_size, since there's no "pool"
b. use QueuePool ( the default) then you can set the max overflow and pool size settings
